File: app/src/androidTest/java/app/we/go/emojidraw/arch/di/TestPracticeModule.java
package app.we.go.emojidraw.arch.di;

import app.we.go.emojidraw.arch.di.qualifier.PracticeDuration;
import app.we.go.emojidraw.arch.di.scope.ActivityScope;
import app.we.go.emojidraw.features.practice.PracticeContract;
import app.we.go.emojidraw.features.practice.StubPracticePresenter;
import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@ActivityScope
@Module
public abstract class TestPracticeModule {

    @Binds
    @ActivityScope
    abstract PracticeContract.Presenter providePracticePresenter(StubPracticePresenter presenter);

    @Provides
    @ActivityScope
    @PracticeDuration
    static Integer providePracticeDuration() {
        return 2;
    }


}

File: app/src/androidTest/java/app/we/go/emojidraw/arch/di/TestPracticeModule.kt
package app.we.go.emojidraw.arch.di

import app.we.go.emojidraw.features.practice.PracticeContract
import app.we.go.emojidraw.features.practice.StubPracticePresenter
import dagger.Binds
import dagger.Module
import dagger.Provides

@Module
abstract class TestPracticeModule {

    @Binds
    @ActivityScope
    internal abstract fun providePracticePresenter(presenter: StubPracticePresenter): PracticeContract.Presenter


    @Module
    companion object {

        @Provides
        @ActivityScope
        @PracticeDuration
        @JvmStatic
        internal fun providePracticeDuration(): Int {
            return 2
        }
    }


}